pkg/build/strategies/sti/sti.go

```diff
@@ -237,27 +243,18 @@ func (b *STI) PostExecute(containerID string, location string) error {
 // Determine determines if the current build supports incremental workflow.
 // It checks if the previous image exists in the system and if so, then it
 // verifies that the save-artifacts script is present.
-func (b *STI) Determine(request *api.Request) (err error) {
-	//request.Incremental = false
-
+func (b *STI) Determine(request *api.Request) bool {
 	if request.Clean {
-		return
+		return false
 	}
 
-	// can only do incremental build if runtime image exists
-	previousImageExists, err := b.docker.IsImageInLocalRegistry(request.Tag)
-	if err != nil {
-		return
+	// can only do incremental build if runtime image exists, so always pull image
+	previousImageExists, _ := b.docker.IsImageInLocalRegistry(request.Tag)
+	if image, _ := b.docker.PullImage(request.Tag); image != nil {
+		previousImageExists = true
 	}
 
-	// we're assuming save-artifacts to exist for embedded scripts (if not we'll
-	// warn a user upon container failure and proceed with a clean build)
-	// for external save-artifacts - check its existence
-	saveArtifactsExists := request.ExternalOptionalScripts ||
-		b.fs.Exists(filepath.Join(request.WorkingDir, "upload", "scripts", string(api.SaveArtifacts)))
-
-	request.Incremental = previousImageExists && saveArtifactsExists
-	return nil
+	return previousImageExists && b.installedScripts[api.SaveArtifacts]
```

Comment:
so if i have an image that includes a save-artifacts script, and i do not provide an external save-artifacts script, this will always return false, won't it?  because installedScripts[api.SaveArtifacts] will never be true.  So we'll never attempt an incremental build for that image, and thus never invoke its built in save-artifacts.

Am I misunderstanding something?